### PR TITLE
BZ-4216 Android build fails when not cross-compiled on Linux

### PR DESCRIPTION
**Pull request by dlifshitz-maca**
[Changed]
platform_android.GNU
- replace the platform_linux_common.GNU include with its contents
  except the erroneous call to getconf

[Tests]
Test 1:
1) Compile ACE for Android using OSX or Windows
2) Verify build is successful

### CODE REVIEW

File: ACE/include/makeinclude/platform_android.GNU [hunk 1]
```diff
@@ -9,7 +9,123 @@ no_hidden_visibility ?= 1
 # as of NDK r6 inlining is required
 inline ?= 1
 
-include $(ACE_ROOT)/include/makeinclude/platform_linux_common.GNU
+# start of: include most of platform_linux_common.GNU
```
Comment (jwillemsen) [inline ACE/include/makeinclude/platform_android.GNU]: No need for this comment to my idea, we have git history to know where changes are coming from

File: ACE/include/makeinclude/platform_android.GNU [hunk 2]
```diff
@@ -9,7 +9,123 @@ no_hidden_visibility ?= 1
...
+
+# We always include config-linux.h on Linux platforms.
+ACE_PLATFORM_CONFIG ?= config-linux.h
```
Comment (jwillemsen) [inline ACE/include/makeinclude/platform_android.GNU]: Shouldn't this be config-android.h now?

File: ACE/include/makeinclude/platform_android.GNU (hunk 1 above)
Comment (dlifshitz-maca) [inline ACE/include/makeinclude/platform_android.GNU]: Fair enough. I added this for my project but wasn't sure if it was needed in the master repo. I'll remove it in my next update when this round of the review is complete.

**Comment by jwillemsen** [comment]
Probably more contents taken from the original file doesn't apply to android, but can imagine it is hard to sort that out

File: ACE/include/makeinclude/platform_android.GNU (hunk 2 above)
Comment (dlifshitz-maca) [inline ACE/include/makeinclude/platform_android.GNU]: I've confirmed that this gets overwritten elsewhere and doesn't get included. I'll remove it too.

**Comment by dlifshitz-maca** [comment]
Agreed that we most likely don't need a lot of the content. I decided to leave it so I don't accidentally break something. I'm not an expert on compiler flags. Please let me know when you have finished the review and I will upload my latest changes.

**Comment by jwillemsen** [comment]
I will not be able to review further at this moment, please upload your latest changes, maybe someone else can review further

**Comment by jwillemsen** [comment]
Can you commit and push your improvements?

**Comment by dlifshitz-maca** [comment]
@jwillemsen Yes, I will put it on my list for next week. Thanks for the reminder.

**Pull request merged**